ui/src/dashboards/utils/time.ts

```diff
@@ -28,3 +33,38 @@ export const millisecondTimeRange = ({
   }
   return {since, until}
 }
+
+const nullTimeRange: TimeRange = {lower: null, upper: null}
+
+const validRelativeTimeRange = (timeRange: TimeRange): TimeRange => {
+  const validatedTimeRange = timeRanges.find(t => t.lower === timeRange.lower)
+
+  if (validatedTimeRange) {
+    return validatedTimeRange
+  }
+
+  return nullTimeRange
+}
+
+export const validAbsoluteTimeRange = (timeRange: TimeRange): TimeRange => {
+  if (timeRange.lower && timeRange.upper) {
```

Comment:
will `moment(null).isValid()` return false? If so, you probably don't need this outer condition.

Reply:
is does, but `moment(undefined).isValid()` returns true, so i suppose i look at this as a more explicit guard against drifting default values